frontend/src/entities/configuration/dynamic-configuration.tsx

```diff
@@ -218,413 +223,520 @@ export function DynamicConfiguration () {
         setFields({ 'configuration.EMRConfig.applications': updatedSelectedApps });
     };
 
-    return (
-        <Container
-            header={
-                <Header
-                    variant='h2'
-                    description={`The current dynamic configuration of ${window.env.APPLICATION_NAME}. These settings can be modified without redeploying the application.`}
-                >
-                    {window.env.APPLICATION_NAME} Dynamic Configuration
-                </Header>
+
+    const isObject = (x) => typeof x === 'object' && !Array.isArray(x) && x !== null;
+
+    /**
+     * Computes the difference between two JSON objects, recursively.
+     *
+     * This function takes two JSON objects as input and returns a new object that
+     * contains the differences between the two. Works with nested objects.
+     *
+     * @param {object} [obj1={}] - The first JSON object to compare.
+     * @param {object} [obj2={}] - The second JSON object to compare.
+     * @returns {object} - A new object containing the differences between the two input objects.
+     */
+    function getJsonDifference (obj1 = {}, obj2 = {}) {
+        const output = {},
+            merged = { ...obj1, ...obj2 }; // has properties of both
+
+        for (const key in merged) {
+            const value1 = obj1[key], value2 = obj2[key];
+
+            if (isObject(value1) || isObject(value2)) {
```

Comment:
This works currently because both objects have the same structure, but this should probably be an `&&` instead of a `||`. This function expects both inputs to be objects and I think would have an error if you attempted `merged = { ...obj1, ...obj2};` if one of them is not a dict/object

Reply:
same as above